apps/server/src/shared/repo/user/user.repo.integration.spec.ts

```diff
@@ -75,4 +75,40 @@ describe('user repo', () => {
 			expect(result.roles[0].roles[0].roles.getItems()).toEqual(roles3);
 		});
 	});
+
+	it('should be defined', () => {
+		expect(repo).toBeDefined();
+		expect(typeof repo.findByLdapId).toEqual('function');
+	});
+
+	describe('findByLdapId', () => {
+		afterEach(async () => {
+			await em.nativeDelete(User, {});
+		});
+
+		it('should return right keys', async () => {
+			const user = userFactory.build();
+
+			await em.persistAndFlush([user]);
+			const result = await repo.findByLdapId(user.ldapId);
+			expect(Object.keys(result).sort()).toEqual(
+				['createdAt', 'updatedAt', 'roles', 'firstName', 'lastName', 'email', 'school', '_id'].sort()
+			);
+		});
+
+		it('should return one role that matched by id', async () => {
```

Comment:
why are you referring to roles here?

Reply:
removed the mention of roles. Now it makes more sense.